sklearn/utils/tests/test_validation.py

```diff
@@ -2358,3 +2359,34 @@ def test_force_all_finite_rename_warning():
 
     with pytest.warns(FutureWarning, match=msg):
         as_float_array(X, force_all_finite=True)
+
+
+@pytest.mark.parametrize(
+    ["X", "estimator", "expectation"],
+    [
+        (
+            np.array([[[1, 2], [3, 4]], [[1, 2], [3, 4]]]),
+            RandomForestRegressor(),
+            pytest.raises(
+                ValueError,
+                match="Found array with dim 3. RandomForestRegressor expected <= 2.",
+            ),
+        ),
+        (
+            np.array([[[1, 2], [3, 4]], [[1, 2], [3, 4]]]),
+            None,
+            pytest.raises(
+                ValueError,
+                match="Found array with dim 3. Expected <= 2.",
+            ),
+        ),
+        (
+            np.array([[1, 2], [3, 4]]),
+            None,
+            does_not_raise(),
+        ),
+    ],
+)
+def test_check_array_allow_nd_errors(X, estimator, expectation) -> None:
+    with expectation:
```

Comment:
We usually use the `with pytest.raises` context manager. For consistence, would you mind doing this here, as well?